source/library/Witch/Instances.hs

```diff
@@ -207,16 +207,9 @@ instance TryFrom.TryFrom Int.Int32 Word where
 instance TryFrom.TryFrom Int.Int32 Natural.Natural where
   tryFrom = Utility.eitherTryFrom fromNonNegativeIntegral
 
--- | Uses 'fromIntegral' when the input is between -16,777,215 and 16,777,215
--- inclusive.
+-- | TODO
 instance TryFrom.TryFrom Int.Int32 Float where
-  tryFrom = Utility.eitherTryFrom $ \s ->
-    if s < -maxFloat
-      then Left Exception.Underflow
-      else
-        if s > maxFloat
-          then Left Exception.Overflow
-          else Right $ fromIntegral s
+  tryFrom = Utility.eitherTryFrom $ TryFrom.tryFrom @Integer . From.from
```

Comment:
These instances probably should've been written this way in the first place. Rather than repeating this logic for each integral type, instead just convert into `Integer` and do the checks there.